base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -818,7 +820,9 @@ where B: BlockchainBackend
         // to increase the timestamp to be greater than the median timestamp otherwise the block wont be accepted by
         // nodes
         if median_timestamp > header.timestamp {
-            header.timestamp = median_timestamp.increase(1);
+            header.timestamp = median_timestamp
+                .checked_add(EpochTime::from(1))
+                .ok_or(ChainStorageError::OutOfRange)?;
```

Comment:
```suggestion
                .ok_or(ChainStorageError::UnexpectedResult("Timestamp calculation overflowed".to_string()))?;
```
The OutofRange error is used when requests are out of range, this is not the case here and this is closer to the actual error.

Reply:
Will update, thanks! I didn't know the best error to use.